critiquebrainz/frontend/external/musicbrainz_db/utils.py

```diff
@@ -47,8 +48,11 @@ def get_entities_by_gids(*, query, entity_type, mbids):
         redirects = query.session.query(redirect_model).filter(redirect_model.gid.in_(remaining_gids))
         redirect_gids = {redirect.redirect_id: redirect.gid for redirect in redirects}
         redirected_entities = query.filter(redirect_model.redirect.property.primaryjoin.left.in_(redirect_gids.keys())).all()
-        for entity in redirected_entities:
-            entity.gid = redirect_gids[entity.id]
```

Comment:
Because these are ORM objects, I wonder if there is a better way of doing this operation so that we don't modify them. What about returning a dictionary of `{gid: entity}` where the `gid` could be a redirect gid, and not just `entity.gid`. This is quite similar to what you're doing in `fetch_multiple_x` anyway, and means that we're not passing around modified but not committed ORM objects.